pytket/pytket/passes/superpass.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2019-2023 Cambridge Quantum Computing
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+from typing import Optional, Callable
+
+from pytket.circuit import Circuit
+from .._tket.passes import BasePass
+
+
+class superpass:
+    """
+    collection of different pytket compilation passes which are
+    all applied on the same circuit. The result of the
+    compilation is best circuit selected by a given metric.
+    """
+
+    def __init__(
+        self,
+        passlist: list[BasePass],
+        score_func: Optional[Callable[[Circuit], int]] = None,
+    ):
+        """
+        Constructs a superpass
+
+        :param passlist: list of pytket compilation passes
+        :param score_func: function to score the
+          results of the compilation, this is optional.
+          The default function is the circuit depth.
```

Comment:
I think it would be better not to have a default but to force the caller to specify a metric. There is no obvious default, and depth will often not be what one wants.

Reply:
Done in [699f5d2](https://github.com/CQCL/tket/pull/1050/commits/699f5d22b7db48558134c65b872f9e71b40d2888)